spec/services/order_syncer_spec.rb

```diff
@@ -146,8 +164,8 @@
           expect(order.bill_address.address1).to eq "123 abc st"
           expect(order.bill_address.phone).to eq "1123581321"
           expect(order.ship_address.firstname).to eq "Bill"
-          expect(order.ship_address.lastname).to eq ship_address_attrs["lastname"]
-          expect(order.ship_address.address1).to eq ship_address_attrs["address1"]
+          expect(order.ship_address.lastname).to eq bill_address_attrs["lastname"]
```

Comment:
why not using the address' object rather than its hash of attributes? I don't see any benefit on using the hash. Just more code.

Reply:
Also, by are we switching from ship to bill address?

Reply:
@sauloperez This is using `_attrs` to keep the original ship and bill attributes after the order is initialized. Syncing modifies the `order` object including sometimes the `original_ship_address` and `original_bill_address`.

Reply:
> Also, why are we switching from ship to bill address? Sorry, I find all this file very hard to follow.

When the shipping method is for a pickup (`require_ship_address == false`), syncing the order with the subscription disregards whatever ship address you have in the subscription. It forces the ship address of the order to be the distributor's address (because it is a pickup) and to use the orders bill contact details.

This is existing behaviour. I didn't actually change the app code here - it just so happened that the previous assertions passed because `order.ship_address`, `order.bill_address`, and `distributor.address` used the same attributes (`create(:address)`).

Reply:
Yes, I remember fixing that behavior for v2. Thanks for the explanation!